framework/src/st_things/things_stack/framework/things_data_manager.c

```diff
@@ -1928,10 +1935,12 @@ int dm_load_legacy_cloud_data(es_cloud_signup_s **cl_data)
 	return ret;
 }
 
-const char *dm_get_things_device_type(int device_id)
+const char *dm_get_things_device_type(const char* device_id)
 {
-	if (device_id != 0)
+	if (strncmp(g_device->device_id, device_id, MAX_DEVICE_ID_LENGTH) != 0) {
```

Comment:
g_device->device_id is NULL;
You did not set this up.
Perhaps this will always return NULL.